### PR TITLE
Update Travis & AppVeyor to allow node 0.10 failures

### PR DESCRIPTION
**Pull request by phated**
I believe this is the correct configuration but I'm still waiting on my AppVeyor to run.

@es128

### CODE REVIEW

**Comment by phated** [comment]
Woot, my AppVeyor had a node 0.10 failure but the suite is marked as successful so it looks like everything is working.

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,14 @@ node_js:
...
   - '0.10'
 os:
   - linux
   - osx
 script:
   - npm run ci-test
 sudo: false
+matrix:
+  allow_failures:
+    - node_js: '0.10'
```
Comment (es128) [inline .travis.yml]: 0.10 is actually reliable enough on Travis. There are random failures on OS X all the time, but Travis makes it easy enough to restart the specific items until they pass. 

I don't think we need this one.

Comment (phated) [reply .travis.yml]: Alright, I'll remove

**Comment by phated** [comment]
Amended and force-pushed.

**Comment by phated** [comment]
@es128 coveralls API is being super flakey so I moved the coverage upload to `after_script` like we do for gulpjs repos

Do you expect v2 to be released today?  I'm really hoping to get gulp4 finalized and published to npm before Jan1

**Comment by es128** [comment]
@phated idk whether it will work out to publish today.

Should we add node v9 to the CI configs in this PR?

**Comment by phated** [comment]
Sure, I'll update shortly. What other things will we be waiting on? I have bandwidth to help out

**Comment by es128** [comment]
Off the top of my head: a new changelog entry and updating the readme appveyor badge to the new paulmillr path

**Comment by phated** [comment]
On it!

**Pull request merged**